system/Helpers/url_helper.php

```diff
@@ -634,3 +634,27 @@ function mb_url_title(string $str, string $separator = '-', bool $lowercase = fa
 }
 
 //--------------------------------------------------------------------
+
+if (! function_exists('url_to'))
+{
+	/**
+	 * Get the full, absolute URL to a controller method
+	 * (with additional arguments)
+	 *
+	 * @param string $controller
+	 * @param mixed  ...$args
+	 *
+	 * @return string
+	 */
+	function url_to(string $controller, ...$args): string
+	{
+		$class = explode('::', $controller)[0];
+
+		if (! class_exists($class))
+		{
+			$controller = service('routes')->getDefaultNamespace() . $controller;
+		}
+
+		return base_url(route_to($controller, ...$args));
```

Comment:
I *think* this needs to be `site_url` to account for subdirectories - can confirm on desktop later.

Reply:
If I change `base_url` to `site_url`, it will prepend `index.php` (so `http://example.com/xyz` would become `http://example.com/index.php/xyz`). Otherwise, it's still fine, so changed that. Will commit soon.

Reply:
The `index.php` should appear or not depending on the user's app setting, which I think is good. It could be another test scenario though, to make sure `url_to()` is respecting `config('App')->indexPage`.

Reply:
No matter what I try, doesn't seem to work, if I try to respect that indexPage setting